syntax = "proto3";

package todo;

message Todo {
  int32 id = 1;
  string title = 2;
  bool completed = 3;
  int64 created_at = 4;
}

message TodoList {
  repeated Todo todos = 1;
}

message CreateTodoRequest {
  string title = 1;
}

message UpdateTodoRequest {
  int32 id = 1;
  bool completed = 2;
}

message DeleteTodoRequest {
  int32 id = 1;
}

message TodoStats {
  int32 total = 1;
  int32 active = 2;
  int32 completed = 3;
}

message ApiResponse {
  bool success = 1;
  string message = 2;
  oneof data {
    Todo todo = 3;
    TodoList todo_list = 4;
    TodoStats stats = 5;
  }
}

// WebSocket real-time events (changes only)
enum EventType {
  CREATED = 0;
  UPDATED = 1;
  DELETED = 2;
}

message TodoEvent {
  EventType type = 1;
  Todo todo = 2;           // For CREATED/UPDATED
  int32 deleted_id = 3;    // For DELETED
  TodoStats stats = 4;     // Updated stats
}
